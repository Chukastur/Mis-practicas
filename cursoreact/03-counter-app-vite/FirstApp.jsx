import isValidProp from 'react-props-check';

export const FirstApp = ( { 
    title = 'Hola Mundo', 
    subTitle = 'Soy Fernando', 
    name = 'Fernando Lopez' 
} ) => {

    if (!isValidProp('title') || typeof title !== 'string') {
        console.error("El prop 'title' debe ser un string y es requerido.");
    }

    if (typeof subTitle !== 'string') {
        console.error("El prop 'subtitle' debe ser un string.");
    };
    
    return (
        <>
        <h1 data-testid = "test-title"> { title } </h1>
        <p>{ subTitle }</p>
        <p>{ name }</p>
        </>
    );
};
