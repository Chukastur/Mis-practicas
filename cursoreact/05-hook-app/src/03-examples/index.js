

export * from './MultipleCustomHooks'
export * from './LoadingMesage'
export * from './PokemonCard'
export * from '../hooks/useCounter'
export * from '../hooks/useFetch'
export * from '../hooks/useForm'

