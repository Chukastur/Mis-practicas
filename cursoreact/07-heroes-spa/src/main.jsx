import { StrictMode, react } from 'react';
import { createRoot } from 'react-dom/client';
import { BrowserRouter, RouterProvider } from "react-router-dom";

import {HeroesApp} from './HeroesApp';


createRoot(document.getElementById('root')).render(
  <StrictMode>
    
   <BrowserRouter>
    <HeroesApp />
  </BrowserRouter>
   
  </StrictMode>
);
