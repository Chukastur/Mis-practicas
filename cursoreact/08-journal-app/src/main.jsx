
import { BrowserRouter } from 'react-router-dom';
import { createRoot } from 'react-dom/client'
import { Provider } from 'react-redux'
import { StrictMode } from 'react'

import { store } from './store'
import { JournalApp } from './JournalApp'
import './styles.css'

createRoot(document.getElementById('root')).render(

  <StrictMode>
    <Provider store={store}>
      <BrowserRouter>
        <JournalApp />
      </BrowserRouter>
    </Provider>
  </StrictMode>

);
