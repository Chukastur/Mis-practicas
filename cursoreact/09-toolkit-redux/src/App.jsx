import { useDispatch, useSelector } from 'react-redux';
import { decrement, increment, incrementBy } from './store';
import './App.css';

function App() {

  const { counter } = useSelector ( state => state.counter);
  const dispatch = useDispatch();

  return (
    <>
      <h1>Vite + React</h1>
      <p>count is { counter }</p>
      <div>
        <button onClick={ () => dispatch ( increment () ) }>Suma</button>
          
        <button onClick={ () => dispatch ( decrement () ) }>Resta</button>

        <button onClick={ () => dispatch ( incrementBy (2) ) }>Suma 2</button> 
      </div>
    </>

  );
};

export default App;
