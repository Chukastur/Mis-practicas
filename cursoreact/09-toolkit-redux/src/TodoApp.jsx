import { useState } from "react";
import { useGetTodoByIdQuery, useGetTodosQuery } from "./store/apis/todosApi";


export const TodoApp = () => {

  const [todoId, setTodoId ] = useState (1);
                                                 //destructuramos y renombramos data
  //const { data: todos = [] , isLoading } = useGetTodosQuery ();
  const { data: todo, isLoading } = useGetTodoByIdQuery (todoId); 
  
  const nextTodo = () => { 
    setTodoId ( todoId + 1 );
  };

  const prevTodo = () => { 
    if ( todoId === 1) return;
    setTodoId ( todoId - 1 );
  };
  

  return (
   <>
      <h1>Todos - RTK Query</h1>
      <hr />
      <h4>isLoading...{ isLoading ? 'Tue' : 'False'}</h4>

      <pre>{ JSON.stringify ( todo )}</pre>

      <button onClick={ prevTodo }>
        Prev Todo
      </button>
      <button onClick={ nextTodo }>
        Next Todo
      </button>
   </>
  );
};
