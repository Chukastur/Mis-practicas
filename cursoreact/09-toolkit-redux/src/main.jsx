import { StrictMode } from 'react';
import { createRoot } from 'react-dom/client';
import { Provider } from 'react-redux';
import { store } from './';
// import App from './App.jsx';
import { PokemonApp } from './PokemonApp.jsx';
import './index.css';
import { TodoApp } from './TodoApp.jsx';



createRoot(document.getElementById('root')).render(
  <StrictMode>

    <Provider store={ store }>
      
      {/* <App />       */}
      {/* <PokemonApp /> */}
      {<TodoApp />}

    </Provider>

  </StrictMode>
);
  