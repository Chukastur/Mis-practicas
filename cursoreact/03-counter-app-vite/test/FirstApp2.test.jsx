import { FirstApp } from '../src/FirstApp';
import { render, screen } from '@testing-library/react';

describe('Pruebas en <FirtsApp />', () => { 

    const title    = 'Hola, Soy Goku';
    const subTitle = 'Soy un subtitulo';

    test ('debe hacer match con el snapshot',  () => {

        const { container } = render ( <FirstApp title = { title } /> );
        expect ( container ).toMatchSnapshot ();
        
    });

    test ('debe mostrar el mensaje "Hola, Soy Goku',  () => {

        render ( <FirstApp title = { title } /> );
        expect ( screen.getByText ( title )).toBeTruthy ();
        
    });

    test('debe de mostrar el texto en un h1', () => {

        render ( <FirstApp title = { title } /> );
        expect ( screen.getByRole ('heading', { level: 1 }).innerHTML).toContain ( title );

    });

    test('debe de mostrar el subtitulo enviado por props', () => {

        render ( 
            <FirstApp 
                title = { title }
                subTitle = { subTitle }    
            /> 
        );
        
        expect ( screen.getAllByText ( subTitle ).length).toBe ( 1 );        

    });

}); 