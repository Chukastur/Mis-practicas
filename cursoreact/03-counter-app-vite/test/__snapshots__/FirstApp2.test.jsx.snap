// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <FirsApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <p>
    Soy Fernando
  </p>
  <p>
    Fernando Lopez
  </p>
</div>
`;

exports[`Pruebas en <FirtsApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <p>
    Soy Fernando
  </p>
  <p>
    Fernando Lopez
  </p>
</div>
`;
